pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -327,6 +327,21 @@ func (a *ClusterDeploymentValidatingAdmissionHook) validateUpdate(admissionSpec
 		}
 	}
 
+	// Check to make sure only allowed fields under controlPlaneConfig are being modified
+	hasChangedImmutableControlPlaneConfigFields := hasChangedImmutableControlPlaneConfigFields(&oldObject.Spec, &newObject.Spec)
```

Comment:
seems like an unnecessary variable: `if hasChangedImmutableControlPlaneConfigFields(...) {`
should be enough